CMakeLists.txt

```diff
@@ -185,6 +183,16 @@ FATAL: Cannot build wpilib without wpimath.
     )
 endif()
 
+if(NOT WITH_WPIMATH AND WITH_GUI)
+    message(
+        FATAL_ERROR
+        "
+FATAL: Cannot build GUI modules without wpimath.
+       Enable wpimath by setting WITH_WPIMATH=ON
+"
+    )
+endif()
+
```

Comment:
At some point it would be good to split this out a bit more.  Several GUI tools require ntcore (glass and outlineviewer) and wpimath (glass and sysid), but many others do not (fieldImages, imgui, wpigui, RTNS, datalogtool).

Reply:
We could make a macro that takes all the libraries a target depends on, and fail if one is missing. That would mean one `WITH_*` option for every target (except for maybe wpiutil as a library that can't be excluded), but I would welcome more granularity here.

Reply:
Do we fail if one is missing, or just force-enable it to the build, or do we force-disable (with an info message) any tools that depend on a de-selected library?  Fundamentally do we want to make it easier to select things or de-select things?  Is erroring out the right behavior on a conflict?

Reply:
If a target is explicitly enabled and its dependent targets weren't given explicit flags, the dependent targets should be auto-enabled. If a target is explicitly enabled and one or more of its dependent targets is explicitly disabled, the configuration step should fail with a message explaining the conflict.

Reply:
We'd have to manually set the value to auto, which makes it less ergonomic since the variable isn't a boolean anymore. I think that a simple macro that looks like `wpilib_check(OPTIONS WITH_WPIMATH WITH_JAVA REQUIRES WITH_WPIUNITS)` which validates the options and displays an error message like `WITH_WPIMATH and WITH_JAVA requires WITH_WPIUNITS to be on` would be good enough.

Reply:
I've implemented the `wpilib_check` macro as mentioned. Error messages now look like this:
```
CMake Error at cmake/modules/OptionValidation.cmake:23 (message):


  FATAL: WITH_SIMULATION_MODULES requires BUILD_SHARED_LIBS.

         Either enable BUILD_SHARED_LIBS or disable WITH_SIMULATION_MODULES.

Call Stack (most recent call first):
  CMakeLists.txt:103 (wpilib_config)
```
Or in the case of multiple options requiring another:
```
CMake Error at cmake/modules/OptionValidation.cmake:21 (message):


  FATAL: WITH_WPIMATH and WITH_JAVA require WITH_WPIUNITS.

         Either enable WITH_WPIUNITS or disable one of the listed options.

Call Stack (most recent call first):
  CMakeLists.txt:111 (wpilib_config)
```